c++/src/kj/compat/http.c++

```diff
@@ -4384,8 +4388,14 @@ private:
     if (statusCode == 204 || statusCode == 205 || statusCode == 304) {
       // No entity-body.
     } else KJ_IF_MAYBE(s, expectedBodySize) {
-      lengthStr = kj::str(*s);
-      connectionHeaders[HttpHeaders::BuiltinIndices::CONTENT_LENGTH] = lengthStr;
+      // HACK: We interpret a zero-length expected body length on responses to HEAD requests to mean
+      //   "don't set a Content-Length header at all." This provides a way to omit a body header on
+      //   HEAD responses with non-null-body status codes. This is a hack that *only* makes sense
+      //   for HEAD responses.
+      if (method != HttpMethod::HEAD || *s > 0) {
```

Comment:
Seems like it would be cleaner to edit the condition on line 4407, below, which is already specific to head requests?

Reply:
Eh I guess not really. Whatever, we'll probably want to find a better answer later anyway.